Outcomes/outcomes.py

```diff
@@ -0,0 +1,163 @@
+import itertools
+import time
+import warnings
+
+import numpy as np
+import pandas as pd
+import scipy
+import tqdm.contrib.concurrent
+
+from SEIR.utils import config 
+import pyarrow.parquet as pq
+import pyarrow as pa
+import pandas as pd
+
+
+def run_delayframe_outcomes(config, setup_name, outdir, scenario_seir, scenario_outcomes, nsim = 1, index=1, n_jobs=1):
+    start = time.monotonic()
+    sim_ids = np.arange(index, index + nsim)
+
+    # Prepare the probability table:
+    # Either mean of probabilities given or from the file... This speeds up a bit the process.
+    # However needs an ordered dict, here we're abusing a bit the spec.
+    config_outcomes = config["outcomes"]["settings"][scenario_outcomes]
+    if (config["outcomes"]["param_from_file"].get()):
+        # load a file from the seir model, to know how to filter the provided csv file
+        sim_id_str = str(sim_ids[0]).zfill(9)
+        diffI = pd.read_parquet(f'model_output/{setup_name}/{sim_id_str}.seir.parquet')
+        diffI = diffI[diffI['comp'] == 'diffI']
+        dates = diffI.time
+        diffI.drop(['comp'], inplace = True, axis = 1)
+
+        # Load the actual csv file
+        branching_file = config["outcomes"]["param_place_file"].as_str()
+        branching_data = pd.read_csv(branching_file, converters={"place": str})
+        branching_data = branching_data[branching_data['place'].isin(diffI.drop('time', axis=1).columns)]
+        if (branching_data.shape[0] != diffI.drop('time', axis=1).columns.shape[0]):
+            raise ValueError(f"Places in seir input files does not correspond to places in outcome probability file {branching_file}")
+
+    parameters = {}
+    for new_comp in config_outcomes:
+        parameters[new_comp] = {}
+        if config_outcomes[new_comp]['source'].exists():
+            # Read the config for this compartement
+            parameters[new_comp]['source'] = config_outcomes[new_comp]['source'].as_str()
+            parameters[new_comp]['probability'] = np.mean(
+                config_outcomes[new_comp]['probability']['value'].as_random_distribution()(size = 10000))
```

Comment:
I think this can be replaced like this:

parameters[new_comp['probability'] = config_outcomes[new_comp]['probability']['value'].as_random_distribution().mean()

Reply:
This doesn't work in the general case in the current way utils.py is written.